test/suite/test_cursor_bound02.py

```diff
@@ -0,0 +1,133 @@
+        self.session.create(uri, create_params)
+        cursor = self.session.open_cursor(uri)
+
+        # Test bound API: Basic usage.
+        cursor.set_key(self.gen_key(40))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+        cursor.set_key(self.gen_key(90))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: Upper bound < lower bound.
+        cursor.set_key(self.gen_key(30))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=upper"), '/Invalid argument/')
+
+        # Test bound API: Lower bound > upper bound.
+        cursor.set_key(self.gen_key(99))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=lower"), '/Invalid argument/')
+
+        # Test bound API: Test resetting lower bound to 20, which would succeed setting upper
+        # bound to 30
+        cursor.set_key(self.gen_key(20))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+        cursor.set_key(self.gen_key(30))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: Test that clearing the lower bound works.
+        cursor.set_key(self.gen_key(10))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=upper"), '/Invalid argument/')
+        self.assertEqual(cursor.bound("action=clear,bound=lower"), 0)
+        cursor.set_key(self.gen_key(10))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: Test that clearing the upper bound works. 
+        cursor.set_key(self.gen_key(20))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=lower"), '/Invalid argument/')
+        self.assertEqual(cursor.bound("action=clear,bound=upper"), 0)
+        cursor.set_key(self.gen_key(20))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+    
+        # Test bound API: Test that clearing both of the bounds works. 
+        cursor.set_key(self.gen_key(50))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+        self.assertEqual(cursor.bound("action=clear"), 0)
+        cursor.set_key(self.gen_key(90))
+        self.assertEqual(cursor.bound("bound=lower"), 0)
+        cursor.set_key(self.gen_key(99))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+
+        # Test bound API: No key set.
+        cursor.reset()
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=lower"), '/Invalid argument/')
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError, lambda: cursor.bound("bound=upper"), '/Invalid argument/')
+        
+        # Test bound API: Test that reset also clears both of the bounds. 
+        cursor.reset()
+        cursor.set_key(self.gen_key(80))
+        self.assertEqual(cursor.bound("bound=upper"), 0)
+        cursor.set_key(self.gen_key(30))
```

Comment:
Can you add some additional cases:

1. A clear followed by a reset
2. Moving bounds, so setting a bound and then changing it.
3. Testing that the key persists, so do a set_key("blah) cursor->bound, followed by a set_value cursor->insert.